packages/ra-ui-materialui/src/input/AutocompleteInput.tsx

```diff
@@ -1,193 +1,63 @@
- * Note that you must also specify the `matchSuggestion` prop
- * @example
- * const choices = [
- *    { id: 123, first_name: 'Leo', last_name: 'Tolstoi' },
- *    { id: 456, first_name: 'Jane', last_name: 'Austen' },
- * ];
- * const matchSuggestion = (filterValue, choice) => choice.first_name.match(filterValue) || choice.last_name.match(filterValue);
- * const FullNameField = ({ record }) => <span>{record.first_name} {record.last_name}</span>;
- * <SelectInput source="gender" choices={choices} optionText={<FullNameField />} matchSuggestion={matchSuggestion} />
- *
- * The choices are translated by default, so you can use translation identifiers as choices:
- * @example
- * const choices = [
- *    { id: 'M', name: 'myroot.gender.male' },
- *    { id: 'F', name: 'myroot.gender.female' },
- * ];
- *
- * However, in some cases (e.g. inside a `<ReferenceInput>`), you may not want
- * the choice to be translated. In that case, set the `translateChoice` prop to false.
- * @example
- * <AutocompleteInput source="gender" choices={choices} translateChoice={false}/>
- *
- * The object passed as `options` props is passed to the material-ui <TextField> component
- *
- * @example
- * <AutocompleteInput source="author_id" options={{ color: 'secondary', InputLabelProps: { shrink: true } }} />
- */
 export const AutocompleteInput = (props: AutocompleteInputProps) => {
     const {
         allowEmpty,
-        className,
-        classes: classesOverride,
-        clearAlwaysVisible,
-        choices = [],
+        choices,
+        create,
         createLabel,
         createItemLabel,
         createValue,
-        create,
-        disabled,
         emptyText,
         emptyValue,
         format,
-        fullWidth,
         helperText,
         id: idOverride,
         input: inputOverride,
-        inputText,
         isRequired: isRequiredOverride,
         label,
-        limitChoicesToValue,
         loaded,
         loading,
-        margin = 'dense',
+        limitChoicesToValue,
         matchSuggestion,
         meta: metaOverride,
         onBlur,
         onChange,
         onCreate,
         onFocus,
-        options: {
-            suggestionsContainerProps,
-            labelProps,
-            InputProps,
-            ...options
-        } = {
-            suggestionsContainerProps: undefined,
-            labelProps: undefined,
-            InputProps: undefined,
-        },
+        options,
         optionText = 'name',
         optionValue = 'id',
         parse,
-        refetch,
-        resettable,
         resource,
         setFilter,
-        shouldRenderSuggestions: shouldRenderSuggestionsOverride,
+        shouldRenderSuggestions,
         source,
         suggestionLimit,
-        translateChoice = true,
+        translateChoice,
         validate,
-        variant = 'filled',
         ...rest
     } = props;
 
-    if (isValidElement(optionText) && !inputText) {
```

Comment:
Why did you remove these warnings?

Reply:
This is the unnecessary breaking change I was talking about. Will be fixed soon